File: src/Blog/Blog.Blazor/Data/Entities/RelationType.cs
namespace Blog.Blazor.Data.Entities;

public sealed class RelationType
{
    public int Id { get; set; }
    public string Key { get; set; } = string.Empty;
    public string Family { get; set; } = string.Empty;
    public bool Symmetric { get; set; }
    public string OutboundLabelEn { get; set; } = string.Empty;
    public string OutboundLabelPtBr { get; set; } = string.Empty;
    public string InboundLabelEn { get; set; } = string.Empty;
    public string InboundLabelPtBr { get; set; } = string.Empty;
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public List<ContentRelation> Relations { get; set; } = [];
}

File: laravel/app/Models/RelationType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class RelationType extends Model
{
    use HasFactory;

    protected $fillable = ['key', 'family', 'symmetric', 'outbound_label_en', 'outbound_label_pt_br', 'inbound_label_en', 'inbound_label_pt_br'];

    /**
     * @return HasMany<ContentRelation, $this>
     */
    public function contentRelations(): HasMany
    {
        return $this->hasMany(ContentRelation::class);
    }
}
